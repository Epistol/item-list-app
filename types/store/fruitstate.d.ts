export interface FruitState {
  data: any[];
  fruitData: any[];
}
