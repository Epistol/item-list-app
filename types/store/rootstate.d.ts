export interface RootState {
  route: any;
}
