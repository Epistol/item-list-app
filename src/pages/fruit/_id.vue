<template>
  <div class="w-full">
    <ViewFruit />
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, reactive, onMounted } from '@vue/composition-api'
import ViewFruit from '~/components/Fruits/ViewFruit.vue'
export default defineComponent({
  name: 'PageFruit',
  components: {
    ViewFruit,
  },
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  setup(props, ctx) {
    onMounted(async () => {})

    const slug = ctx.root.$route.params?.id

    onMounted(async () => {
      await ctx.root.$store.dispatch('fruit/getFruitData', slug)
    })

    return { slug }
  },
})
</script>
