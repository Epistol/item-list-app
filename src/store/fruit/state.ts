// import { FruitState } from "@@/types";

const state = (): any => ({
  data: [],
  fruitData: []
});

export default state;
