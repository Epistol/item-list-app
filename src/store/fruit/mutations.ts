import { MutationTree } from "vuex";
import { FruitState } from "@@/types/store/fruitstate";
import _ from "lodash";
import { Fruit } from "../../../types/fruit";

const mutations: MutationTree<FruitState> = {
  SET_DATA(state, data: any) {
    // For each entry, we'll check if it contain an object
    let goodFruits: Fruit[] = [];

    const addFruit = (data: any) => {
      if (_.isPlainObject(data)) {
        if (_.has(data, "description")) {
          // It's a fruit
          goodFruits.push(data);
        } else {
          // It has another object in it
          // (we don't check the other data, they are not with a description)
          Object.entries(data).map((element: any) => {
            addFruit(element[1]);
          });
        }
      }
    };

    Object.entries(data.data).map((value, index): any => {
      // If it is an object with data in it (or another object)
      addFruit(value[1]);
      // If it's one the fruits that we added, we need to search in the array (not a plain object)
      if (value[0] === "more") {
        const addedFruits: any = value[1];

        addedFruits.map((element: any) => {
          if (_.has(element, "description")) {
            // It's a fruit
            goodFruits.push(element);
          }
        });
      }
    });

    state.data = goodFruits;
  },

  SET_FRUIT_DATA(state, data: any) {
    // For each entry, we'll check if it contain an object
    state.fruitData = data;
  },

  DELETE_FRUIT(state, id: any) {
    state.data = state.data.filter((item: any) => item.id !== id);
  }
};

export default mutations;
