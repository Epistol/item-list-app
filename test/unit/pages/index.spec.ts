import { shallowMount, createLocalVue } from "@vue/test-utils";
import VueCompositionApi from "@vue/composition-api";
import Vuex, { Store } from "vuex";
import { RootState } from "@@/types/store/rootstate";
import stubs from "@@/test/mocks/stubs";
import mocks from "@@/test/mocks/mocks";
import globalStore from "@@/test/mocks/store";
import { fruitList } from "@@/test/mocks/api/fruit";

import Index from "@/pages/index.vue";

const localVue = createLocalVue();

localVue.use(VueCompositionApi);
localVue.use(Vuex);

globalStore.modules.fruit.getters.data = jest.fn(() => fruitList);

describe("Index.vue", () => {
  let store: Store<RootState>;
  let wrapper: any;
  let options: any;

  beforeEach(() => {
    store = new Vuex.Store(globalStore);
    mocks.$store = store;
    options = { localVue, stubs, mocks, store };
    wrapper = shallowMount(Index, options);
  });

  describe("component", () => {
    it("exists", () => {
      expect(wrapper.exists()).toBe(true);
    });

    it("renders correctly", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
