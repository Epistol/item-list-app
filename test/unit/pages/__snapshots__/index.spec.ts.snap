// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index.vue component renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
